src/python/pants/bin/local_pants_runner.py

```diff
@@ -277,5 +279,6 @@ def run(self, start_time: float) -> ExitCode:
                 metrics = self.graph_session.scheduler_session.metrics()
                 self.run_tracker.set_pantsd_scheduler_metrics(metrics)
                 self.run_tracker.end_run(engine_result)
+                self.graph_session.scheduler_session.wait_for_tail_tasks(self.global_bootstrap_options.session_end_tasks_timeout)
```

Comment:
Should put this before the request for metrics and `end_run`, otherwise any metrics generated by the tail tasks will not be captured (cache writes, for example).

Reply:
Will do.

Reply:
Moved it to the start of the `finally` clause.